src/NHibernate/Linq/Visitors/MemberExpressionJoinDetector.cs

```diff
@@ -18,21 +19,26 @@ internal class MemberExpressionJoinDetector : RelinqExpressionVisitor
 	{
 		private readonly IIsEntityDecider _isEntityDecider;
 		private readonly IJoiner _joiner;
+		private readonly ISessionFactoryImplementor _sessionFactory;
 
 		private bool _requiresJoinForNonIdentifier;
 		private bool _preventJoinsInConditionalTest;
 		private bool _hasIdentifier;
 		private int _memberExpressionDepth;
 
-		public MemberExpressionJoinDetector(IIsEntityDecider isEntityDecider, IJoiner joiner)
+		public MemberExpressionJoinDetector(
+			IIsEntityDecider isEntityDecider,
+			IJoiner joiner,
+			ISessionFactoryImplementor sessionFactory)
```

Comment:
This addition seems to have no use, is this some forgotten "trash change"?

Reply:
My bad, `sessionFactory` is used in #2079 for `ExpressionKeyVisitor.Visit` method which is not included in this PR. Removed the unused parameter and field.